web-common/src/main/java/org/openmrs/eip/web/contoller/BaseRestController.java

```diff
@@ -52,6 +52,45 @@ public Object doGet(Long id) {
 		    "jpa:" + getName() + "?query=SELECT c FROM " + getName() + " c WHERE c.id = " + id, null, getClazz());
 	}
 	
+	public Map<String, Object> getByDateCreated(String startDate, String endDate) {
+
+	        Map<String, Object> results = new HashMap<String, Object>(2);
+	        List<Object> items;
+
+	        if (startDate.isBlank() && endDate.isBlank()) {
+	            items = on(camelContext).to("jpa:" + getName() + "?query=SELECT c FROM " + getName()
+	                    + " c &maximumResults=" + DEFAULT_MAX_COUNT).request(List.class);
+	        }
+	        else if (!startDate.isBlank() && endDate.isBlank()) {
+	            
+	            items = on(camelContext).to("jpa:" + getName() + "?query=SELECT c FROM " + getName()
+                   + " c  where c.dateCreated >= '" + startDate
+                   + "' &maximumResults=" + DEFAULT_MAX_COUNT).request(List.class);
+	            
+	        } else if (startDate.isBlank() && !endDate.isBlank()) {
+	         
+	            items = on(camelContext).to("jpa:" + getName() + "?query=SELECT c FROM " + getName()
+                    + " c  where c.dateCreated <= '" + endDate 
+                    + "' &maximumResults=" + DEFAULT_MAX_COUNT).request(List.class);
+	        }
+	        else {
+	            items = on(camelContext).to("jpa:" + getName() + "?query=SELECT c FROM " + getName()
+	                    + " c  where c.dateCreated >= '" + startDate + "' and c.dateCreated <= '"
+	                    + endDate + "' &maximumResults=" + DEFAULT_MAX_COUNT).request(List.class);
+	        }
+
+	        if (items.size() > 0) {
```

Comment:
This if clause is redundant, replace it with just 2 lines, just set the count and set the items with no if clause

Reply:
Done

Reply:
Please remove this if clause, it is redundant

Reply:
Something is not right in this method, you need to get the count first, if count is  > 0, load items otherwise don't, In theory you should generate the where clause once to get both the count vs items, you only alter the parts of the query before and after the where clauses

Reply:
Done